src/components/Button.js

```diff
@@ -46,26 +58,45 @@ export const Button = ({ disabled, invert, notFullWidth, onPress, title }) => {
       titleStyle={[
         styles.titleStyle,
         invert && styles.titleStyleInvert,
-        needLandscapeStyle && styles.titleStyleLandscape
+        needLandscapeStyle && styles.titleStyleLandscape,
+        extraLarge && styles.extraLargeTitle,
+        large && styles.largeTitle,
+        medium && styles.mediumTitle,
+        small && styles.smallTitle
       ]}
       disabledStyle={styles.buttonStyleDisabled}
       disabledTitleStyle={styles.titleStyle}
       buttonStyle={[
         styles.buttonStyle,
         invert && styles.buttonStyleInvert,
-        isDelete && styles.rejectStyle
+        isDelete && styles.rejectStyle,
+        extraLarge && styles.extraLarge,
+        large && styles.large,
+        medium && styles.medium,
+        small && styles.small
       ]}
       containerStyle={[styles.containerStyle, needLandscapeStyle && styles.containerStyleLandscape]}
       ViewComponent={invert || isDelete || disabled ? undefined : DiagonalGradient}
       useForeground={!invert}
       accessibilityLabel={`${title} ${consts.a11yLabel.button}`}
       disabled={disabled}
+      icon={{
+        name: iconName,
+        type: IconSetName.toLowerCase(),
+        containerStyle: {},
+        color: invert ? colors.primary : colors.lightestText,
+        size: small ? normalize(16) : normalize(24)
```

Comment:
icon size does not change for other sizes? shouldn't it depend on text size somehow?

Reply:
the size of the icon is reduced only when the `small` prop is `true`.

Reply:
asking again: shouldn't it depend on text size somehow?

Reply:
what kind of dependency are you talking about? i can use `iconStyle` prop for icon and add different styles for all different sizes.

```
icon={{
  name: iconName,
  type: IconSetName,
  containerStyle: {},
  iconStyle: [
    styles.iconStyle,
    invert && styles.iconStyleInvert,
    extraLarge && styles.extraLargeIcon,
    small && styles.smallIcon,
    smallest && styles.smallestIcon
  ]
}}
```

Reply:
i mean for the `size` prop: if text is standard, icon should be standard. if text is small, icon should be smaller as well? if text is large icon should be larger as well?

Reply:
we do not need the `size` prop if we pass the icon as `<Icon...`, right?

Reply:
I think this problem has been eliminated with the newly updated icon prop.

https://github.com/smart-village-solutions/smart-village-app-app/commit/5933f8013e0cfaaab2a97c661ecfa3b13eda8857